services/blockstorage/service.go

```diff
@@ -62,12 +63,91 @@ func (s *service) CommitBlock(input *services.CommitBlockInput) (*services.Commi
 	return nil, nil
 }
 
+func (s *service) lookForTransactionsBlockHeader(height primitives.BlockHeight) (*services.GetTransactionsBlockHeaderOutput, error) {
+	txBlock, err := s.persistence.GetTransactionsBlock(height)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &services.GetTransactionsBlockHeaderOutput{
+		TransactionsBlockProof: txBlock.BlockProof,
+		TransactionsBlockHeader: txBlock.Header,
+		TransactionsBlockMetadata: txBlock.Metadata,
+	}, nil
+}
+
 func (s *service) GetTransactionsBlockHeader(input *services.GetTransactionsBlockHeaderInput) (*services.GetTransactionsBlockHeaderOutput, error) {
-	panic("Not implemented")
+	if input.BlockHeight > s.lastCommittedBlockHeight && input.BlockHeight - s.lastCommittedBlockHeight <= 5 {
+		c := make(chan *services.GetTransactionsBlockHeaderOutput)
+
+		go func() {
+			// TODO extract to a config
+			const interval = 10
+			const timeout = 10000
+
+			for i:=0; i < timeout; i+= interval {
+				if input.BlockHeight <= s.lastCommittedBlockHeight {
+					lookupResult, err := s.lookForTransactionsBlockHeader(input.BlockHeight)
+
+					if err == nil {
+						c <- lookupResult
+						return
+					}
+				}
+
+				time.Sleep(interval)
+			}
+		}()
+
+		result:= <-c
+		return result, nil
+	}
+
+	return s.lookForTransactionsBlockHeader(input.BlockHeight)
 }
 
-func (s *service) GetResultsBlockHeader(input *services.GetResultsBlockHeaderInput) (*services.GetResultsBlockHeaderOutput, error) {
-	panic("Not implemented")
+func (s *service) lookForResultsBlockHeader(height primitives.BlockHeight) (*services.GetResultsBlockHeaderOutput, error) {
+	txBlock, err := s.persistence.GetResultsBlock(height)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &services.GetResultsBlockHeaderOutput{
+		ResultsBlockProof: txBlock.BlockProof,
+		ResultsBlockHeader: txBlock.Header,
+	}, nil
+}
+
+func (s *service) GetResultsBlockHeader(input *services.GetResultsBlockHeaderInput) (result *services.GetResultsBlockHeaderOutput, err error) {
+	if input.BlockHeight > s.lastCommittedBlockHeight && input.BlockHeight - s.lastCommittedBlockHeight <= 5 {
+		c := make(chan *services.GetResultsBlockHeaderOutput)
+
+		go func() {
```

Comment:
this is the exact same functionality as in `GetTransactionsBlockHeader`, right? if so, consider extracting to a common local function that supports this "wait till block is committed" behavior